### PR TITLE
:gear: Adds pre-commit and `just lint` task

### PR DESCRIPTION
**Pull request by jefftriplett**
TODO: Add `pyproject.toml` before we merge it and probably hold off until the bulk of Drew's schedule work is ready. 

I have had this locally for a while but never pushed it. It needs a `pyproject.toml` to complete it and probably some minor changes to that it doesn't try to change everything. 

I'll probably backport it to the last few years once it's ready.

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (drewbrew) [inline .pre-commit-config.yaml:2]:
```suggestion
  python: python3.11
```

**Review by drewbrew — APPROVED** [review]
I like the concept for sure (if not the ancient python version 😆 )

**Comment by jefftriplett** [comment]
TODO: Add pyproject.toml before we merge it and hold off on running lint until @drewbrew's work is done.

**Pull request merged**
